File: Speaking/Speaking/ViewController.m
//
//  ViewController.m
//  Speaking
//
//  Created by Bobby Towers on 1/13/15.
//  Copyright (c) 2015 Bobby Towers. All rights reserved.
//

#import "ViewController.h"
#import "MyView.h"

@interface ViewController () <MyViewDelegate>

@end

@implementation ViewController
{
    
    MyView * view1;
    
}
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    view1 = [[MyView alloc] init];
 
    // as the VC, I'm going to be your delegate
    view1.delegate = self;
    
    view1.name = @"Bob";

    NSLog(@"view1 name = %@", view1.name);
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

// this method won't show up until you include <MyViewDelegate> protocol above
- (void)changeMyNameTo:(NSString *)name {
    
    NSLog(@"%@", name);

//    view1.name = name;  // view1.name = @"Bob" would create an infinite loop between two methods
                                    // between MyView setName and ViewController changeMyNameTo
    
    NSLog(@"view1 new name = %@", view1.name);
    
}

@end

File: ScrollViewPractice/ScrollViewPractice/ViewController.swift
//
//  ViewController.swift
//  ScrollViewPractice
//
//  Created by Bobby Towers on 3/13/15.
//  Copyright (c) 2015 Bobby Towers. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet var scrollView : UIScrollView!
    @IBOutlet var contentView : UIView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    override func viewDidLayoutSubviews()
    {
        let scrollViewBounds = scrollView.bounds
        let containerViewBounds = contentView.bounds
        
        var scrollViewInsets = UIEdgeInsetsZero
        scrollViewInsets.top = scrollViewBounds.size.height/2.0;
        scrollViewInsets.top -= contentView.bounds.size.height/2.0;
        
        scrollViewInsets.bottom = scrollViewBounds.size.height/2.0
        scrollViewInsets.bottom -= contentView.bounds.size.height/2.0;
        scrollViewInsets.bottom += 1
        
        scrollView.contentInset = scrollViewInsets
    }
}

